flight/Modules/Battery/battery.c

```diff
@@ -149,14 +155,21 @@ static void batteryTask(void * parameters)
 					scaled_voltage = 0.0f;
 					adc_offset_invalid = true;
 				}
+				else if(scaled_voltage > 3.0f && !cells_calculated && (batterySettings.MaxCellVoltage > 0)){
+					cells_calculated = true;
+					cells = ((scaled_voltage / batterySettings.MaxCellVoltage) + 0.9f);
+					flightBatteryData.DetectedCellCount = cells;
+				}
+				else if(!cells_calculated)
+					cells = 1;
 			}
 
 			flightBatteryData.Voltage = scaled_voltage;
 
 			// generate alarms and warnings
-			if (flightBatteryData.Voltage < batterySettings.VoltageThresholds[FLIGHTBATTERYSETTINGS_VOLTAGETHRESHOLDS_ALARM])
+			if (flightBatteryData.Voltage < (batterySettings.VoltageThresholds[FLIGHTBATTERYSETTINGS_VOLTAGETHRESHOLDS_ALARM] * cells))
```

Comment:
Not quite sure what you mean? The autodetection is active as soon as MaxCellVoltage > 0.

Renaming the warning thresholds sounds good. Did not thought about that because i wanted to make this feature off by default so that would have been a non-issue. But now it´s on by default and people who import old values will get bogus detection.